azurerm_backup_policy_vm hourly frequency doesn't validate that hour_duration and hour_interval are present before applying

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.2.6

### AzureRM Provider Version

3.16.0

### Affected Resource(s)/Data Source(s)

azurerm_backup_policy_vm

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "3.16.0"
    }
  }
}

provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "rg" {
  name     = "rg"
  location = "eastus"
}

resource "azurerm_recovery_services_vault" "rsv" {
  name                = "rsv"
  location            = "eastus"
  resource_group_name = azurerm_resource_group.rg.name
  sku                 = "Standard"
}

resource "azurerm_backup_policy_vm" "hourly" {
  name                = "hourly"
  resource_group_name = azurerm_resource_group.rg.name
  recovery_vault_name = azurerm_recovery_services_vault.rsv.name
  policy_type         = "V2"

  backup {
    frequency     = "Hourly"
    time          = "21:00"
  }

  retention_daily {
    count = 10
  }
}
```


### Debug Output/Panic Output

```shell
azurerm_backup_policy_vm.hourly: Creating...
╷
│ Error: Request cancelled
│ 
│   with azurerm_backup_policy_vm.hourly,
│   on main.tf line 26, in resource "azurerm_backup_policy_vm" "hourly":
│   26: resource "azurerm_backup_policy_vm" "hourly" {
│ 
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵

Stack trace from the terraform-provider-azurerm_v3.16.0_x5 plugin:

panic: runtime error: integer divide by zero

goroutine 117 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/recoveryservices.expandBackupProtectionPolicyVMSchedule(0x6820ca0?, {0xc001a65e18, 0x1, 0x1})
        github.com/hashicorp/terraform-provider-azurerm/internal/services/recoveryservices/backup_policy_vm_resource.go:356 +0x911
github.com/hashicorp/terraform-provider-azurerm/internal/services/recoveryservices.resourceBackupProtectionPolicyVMCreateUpdate(0xc00026fd00, {0x572b400?, 0xc001f1e000})
        github.com/hashicorp/terraform-provider-azurerm/internal/services/recoveryservices/backup_policy_vm_resource.go:139 +0x9f8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x68303a8?, {0x68303a8?, 0xc0015d2d20?}, 0xd?, {0x572b400?, 0xc001f1e000?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/resource.go:695 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0010b6fc0, {0x68303a8, 0xc0015d2d20}, 0xc001aa00d0, 0xc00026fb80, {0x572b400, 0xc001f1e000})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/resource.go:837 +0xa7a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0004e80d8, {0x68303a8?, 0xc0015d2c60?}, 0xc000b29860)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.18.0/helper/schema/grpc_provider.go:1021 +0xe3c
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000765040, {0x68303a8?, 0xc0015d24e0?}, 0xc000606620)
        github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/tf5server/server.go:813 +0x4fc
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x5cecee0?, 0xc000765040}, {0x68303a8, 0xc0015d24e0}, 0xc001a1b380, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000dca80, {0x6841c08, 0xc0016bcd00}, 0xc001a90900, 0xc0016b4a80, 0xa87c780, 0x0)
        google.golang.org/grpc@v1.47.0/server.go:1283 +0xcfd
google.golang.org/grpc.(*Server).handleStream(0xc0000dca80, {0x6841c08, 0xc0016bcd00}, 0xc001a90900, 0x0)
        google.golang.org/grpc@v1.47.0/server.go:1620 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.47.0/server.go:922 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.47.0/server.go:920 +0x28a

Error: The terraform-provider-azurerm_v3.16.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Expected Behaviour

### Both hour_interval and hour_duration are missing:
The AzureRM provider error and notify you that `hour_interval` and `hour_duration` are required when frequency is set to `Hourly` for both `terraform plan` and `terraform apply`.

### hour_interval is missing but hour_duration is provided
The AzureRM provider error and notify you that `hour_interval` is required when frequency is set to `Hourly` for both `terraform plan` and `terraform apply`.

### hour_duration is missing but hour_interval is provided
The AzureRM provider error and notify you that `hour_duration` is required when frequency is set to `Hourly` for both `terraform plan` and `terraform apply`.

### Actual Behaviour

### Both hour_interval and hour_duration are missing:
During a `terraform plan` it says the configuration is valid. During a `terraform apply` it says the configuration is valid but panics when attempting to create the backup_policy_vm with an `integer divide by zero` error. I believe this is due to the validation for `hour_duration` being a multiplier of `hour_interval`.

### hour_interval is missing but hour_duration is provided
During a `terraform plan` it says the configuration is valid. During a `terraform apply` it says the configuration is valid but panics when attempting to create the backup_policy_vm with an `integer divide by zero` error. I believe this is due to the validation for `hour_duration` being a multiplier of `hour_interval`.

### hour_duration is missing but hour_interval is provided
During a `terraform plan` it says the configuration is valid. During a `terraform apply` it says the configuration is valid but when attempting to create the backup_policy_vm an error is quickly returned from Azure stating "Input for create or update policy is not in proper format."

### Steps to Reproduce

terraform init
terraform apply

### Important Factoids

N/A

### References

_No response_

--- a/internal/services/recoveryservices/backup_policy_vm_resource.go
+++ b/internal/services/recoveryservices/backup_policy_vm_resource.go
@@ -353,6 +353,16 @@ func expandBackupProtectionPolicyVMSchedule(d *pluginsdk.ResourceData, times []d
 					return nil, fmt.Errorf("`hour_duration` must be specified when `backup.0.frequency` is `Hourly`")
 				}
 
+				if interval == 0 && duration == 0 {
+					return nil, fmt.Errorf("`hour_interval` and `hour_duration` must be specified when `backup.0.frequency` is `Hourly`")
+				}
+				if interval == 0 {
+					return nil, fmt.Errorf("`hour_interval` must be specified when `backup.0.frequency` is `Hourly`")
+				}
+				if duration == 0 {
+					return nil, fmt.Errorf("`hour_duration` must be specified when `backup.0.frequency` is `Hourly`")
+				}
+
 				if duration%interval != 0 {
 					return nil, fmt.Errorf("`hour_duration` must be multiplier of `hour_interval`")
 				}

